src/handlers/scheduledEventHandler.ts

```diff
@@ -1,8 +1,60 @@
+import { KVNamespace } from '@cloudflare/workers-types';
+
 import { handleConfig } from '../config/config';
-import { env } from '../types/global.types';
+import { NAMESPACE_NAME } from '../constants';
+import { env, nicknameUpdateResponseType } from '../types/global.types';
+import { generateJwt } from '../utils/generateJwt';
 
 export async function ping(env: env) {
 	const url = handleConfig(env);
 	const response = await fetch(`${url.baseUrl}/healthcheck`);
 	return response;
 }
+
+export async function callDiscordNicknameBatchUpdate(env: env) {
+	const namespace = env[NAMESPACE_NAME] as unknown as KVNamespace;
+	let lastNicknameUpdate: string | number | null = 0;
+	try {
+		lastNicknameUpdate = await namespace.get('DISCORD_NICKNAME_CHANGED');
+		if (lastNicknameUpdate === null) {
+			throw new Error('Error while fetching KV "DISCORD_NICKNAME_CHANGED" timestamp');
+		}
+	} catch (err) {
+		console.log('Error while fetching the timestamp for last nickname update');
```

Comment:
In the log both `console.log` and `console.error` appears as a generic log. But I've changed it to `console.error` anyway now.

We can also log the `err` so that more details are present. But any of the manual errors being thrown, I've made sure to make them as descriptive as possible. I've added the details for the error in the log as well.